awx/ui_next/src/screens/Template/shared/SurveyList.jsx

```diff
@@ -1,48 +1,230 @@
-    result: questions,
+    result: { questions, name, description },
     error: contentError,
     isLoading,
     request: fetchSurvey,
   } = useRequest(
     useCallback(async () => {
       const {
-        data: { spec },
+        data: { spec = [], description: surveyDescription, name: surveyName },
       } = await JobTemplatesAPI.readSurvey(template.id);
-
-      return spec.map((s, index) => ({ ...s, id: index }));
-    }, [template.id])
+      return {
+        questions: spec.map((s, index) => ({ ...s, id: index })),
+        description: surveyDescription,
+        name: surveyName,
+      };
+    }, [template.id]),
+    { questions: [], name: '', description: '' }
   );
 
   useEffect(() => {
     fetchSurvey();
   }, [fetchSurvey]);
-  if (contentError) {
-    return <ContentError error={contentError} />;
-  }
-  if (isLoading) {
-    return <ContentLoading />;
-  }
 
-  return (
-    questions?.length > 0 &&
-    questions.map((question, index) => (
+  const isAllSelected =
+    selected.length === questions?.length && selected.length > 0;
+
+  const {
+    isLoading: isDeleteLoading,
+    deleteItems: deleteQuestions,
+    deletionError,
+    clearDeletionError,
+  } = useDeleteItems(
+    useCallback(async () => {
+      if (isAllSelected) {
+        return JobTemplatesAPI.destroySurvey(template.id);
+      }
+      const surveyQuestions = [];
+      questions.forEach(q => {
+        if (!selected.some(s => s.id === q.id)) {
+          surveyQuestions.push(q);
+        }
+      });
+      return JobTemplatesAPI.updateSurvey(template.id, {
+        name,
+        description,
+        spec: surveyQuestions,
+      });
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selected]),
+    {
+      qsConfig: QS_CONFIG,
+      fetchItems: fetchSurvey,
+    }
+  );
+  const {
+    isToggleLoading,
+    error: toggleError,
+    request: toggleSurvey,
+  } = useRequest(
+    useCallback(async () => {
+      await JobTemplatesAPI.update(template.id, {
+        survey_enabled: !surveyEnabled,
+      });
+      return setSurveyEnabled(!surveyEnabled);
+    }, [template, surveyEnabled]),
+    template.survey_enabled
+  );
+
+  useEffect(() => {
+    if (toggleError) {
+      setShowToggleError(true);
+    }
+  }, [toggleError]);
+
+  const handleSelectAll = isSelected => {
+    setSelected(isSelected ? [...questions] : []);
+  };
+
+  const handleSelect = item => {
+    if (selected.some(s => s.id === item.id)) {
+      setSelected(selected.filter(s => s.id !== item.id));
+    } else {
+      setSelected(selected.concat(item));
+    }
+  };
+
+  const handleDelete = async () => {
+    await deleteQuestions();
+    setIsDeleteModalOpen(false);
+    setSelected([]);
+  };
+  const canEdit = template.summary_fields.user_capabilities.edit;
+  const canDelete = template.summary_fields.user_capabilities.delete;
+
+  let content;
+  if (
+    (isLoading || isToggleLoading || isDeleteLoading) &&
+    questions?.length <= 0
```

Comment:
I'm curious about this line. What is it for? We only want to show `ContentLoading` if there's no questions? :thinking:

Reply:
I think that was a hold over from me trying different things.  I have removed it.